specification/under_development/profiles/Base_Profile.md

```diff
@@ -0,0 +1,262 @@
+type. The signed integer that is case must be in the interval `[0, nrQubits)`
+for `%Qubit*` and `[0, nrResults)` for `%Result*`, where `nrQubits` and
+`nrResults` are the required number of qubits and results defined by entry point
+attributes.
+
+Only the `%Qubit` and `%Result` data types are required to be supported by all
+backends.
+
+Integers and double-precision floating point numbers are available as in full
+QIR; however, computations using these numeric types are not available.
+
+### Qubit and Result Usage
+
+runtime has autonomy over how to represent qubits. also for base profile, the
+runtime is the only entity that needs to know how to interpret the opaque
+pointers (deref or not). that info is captured in the target triple/duo.
+
+Qubits may not be used after they have been measured. Qubits and results need to
+be numbered consecutively, starting at 0. No qpu support needed for measuring
+individual qubits, but backend would support output selection?
+
+## Quantum Instruction Set
+
+For a Quantum Instruction Set to be compatible with the Base Profile, it needs
+to satisfy the following requirements:
+
+- Since the Base Profile doesn't permit to define local variables, all
+  instructions are required to return void.
+
+## Output Recording
+
+Log format is a separate spec. What the i8* can be is a separate spec. Default
+spec for front-ends to compile into. TODO: do string labels need to be zero
+terminated?
+
+The following functions are declared and used to record the program output: |
+Function                  | Signature         | Description |
+|---------------------------|-------------------|-------------| |
+__quantum__rt__initialize_record_output    | `void(i8*)`  | Inserts a marker in
+the output log that contains an identifier for the used labeling scheme. The
+backend may choose which output format to use, and the label identifier is
+omitted for output formats that do not support labeling. | |
+__quantum__rt__tuple_record_output    | `void(i64, i8*)`  | Inserts a marker in
+the output log that indicates the start of a tuple and how many tuple elements
+are going to be logged. The second parameter reflects an label for the tuple.
+The backend may choose which output format to use. Depending on the used format,
+the label will be logged or omitted. | | __quantum__rt__array_record_output    |
+`void(i64, i8*)`  | Inserts a marker in the output log that indicates the start
+of an array and how many array elements are going to be logged. The second
+parameter reflects an label for the array. The backend may choose which output
+format to use. Depending on the used format, the label will be logged or
+omitted. | |__quantum__rt__result_record_output   | `void(%Result*, i8*)`  |
+Adds a measurement result to the output log. The second parameter reflects an
+label for the result value. The backend may choose which output format to use.
+Depending on the used format, the label will be logged or omitted. |
```

Comment:
We should try and format this markdown table to be more human readable in `.md` form using a MD table formatting tool.

Reply:
On second note it is also rendering incorrectly on Github.

Reply:
Done.